middleware/markdown/process.go

```diff
@@ -17,9 +18,14 @@ const (
 	DefaultStaticDir = "generated_site"
 )
 
+type MarkdownData struct {
+	middleware.Context
+	Doc map[string]interface{}
```

Comment:
I wonder if this could be `map[string]string` - do we ever store anything other than strings?